impl/src/main/java/org/jboss/weld/manager/BeanManagerImpl.java

```diff
@@ -1563,6 +1568,73 @@ public WeldInstance<Object> createInstance() {
+        String loggingMethodName = "isMatchingBean()";
+        if (beanTypes == null || beanQualifiers == null || requiredType == null || requiredQualifiers == null) {
+            throw BeanManagerLogger.LOG.assignabilityMethodIllegalArgs(loggingMethodName);
+        }
+        validateQualifiers(beanQualifiers, loggingMethodName);
+        validateQualifiers(requiredQualifiers, loggingMethodName);
+        // always add Object as bean type; filter bean types to only contain legal bean types
+        Set<Type> legalBeanTypes = new HashSet<>(beanTypes);
+        legalBeanTypes.add(Object.class);
+        legalBeanTypes = Beans.getLegalBeanTypes(legalBeanTypes, beanTypes);
+        // type check first
+        if (BeanTypeAssignabilityRules.instance().matches(requiredType, legalBeanTypes)) {
+            // normalize qualifiers, adding built-ins where missing, then compare
+            Set<Annotation> normalizedBeanQualifiers = Bindings.normalizeBeanQualifiers(beanQualifiers);
+            Set<Annotation> normalizedRequiredQualifiers = requiredQualifiers.isEmpty() ? Set.of(Default.Literal.INSTANCE)
+                    : requiredQualifiers;
+            MetaAnnotationStore metaAnnotationStore = services.get(MetaAnnotationStore.class);
+            if (Beans.containsAllQualifiers(QualifierInstance.of(normalizedRequiredQualifiers, metaAnnotationStore),
+                    QualifierInstance.of(normalizedBeanQualifiers, metaAnnotationStore))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isMatchingEvent(Type eventType, Set<Annotation> eventQualifiers, Type observedEventType,
+            Set<Annotation> observedEventQualifiers) {
+        String loggingMethodName = "isMatchingEvent()";
+        if (eventType == null || eventQualifiers == null || observedEventType == null || observedEventQualifiers == null) {
+            throw BeanManagerLogger.LOG.assignabilityMethodIllegalArgs(loggingMethodName);
+        }
+        if (Types.containsTypeVariable(Types.getCanonicalType(eventType))) {
+            throw BeanManagerLogger.LOG.eventTypeUnresolvableWildcard(eventType);
+        }
+        validateQualifiers(eventQualifiers, loggingMethodName);
+        validateQualifiers(observedEventQualifiers, loggingMethodName);
+        // type check first
+        if (EventTypeAssignabilityRules.instance().matches(observedEventType,
+                HierarchyDiscovery.forNormalizedType(eventType).getTypeClosure())) {
+            // normalize qualifiers, adding built-ins where missing, then compare
+            Set<Annotation> normalizedEventQualifiers = new HashSet<>(eventQualifiers);
+            if (eventQualifiers.isEmpty()
+                    || (eventQualifiers.size() == 1 && eventQualifiers.iterator().next().equals(Any.Literal.INSTANCE))) {
+                normalizedEventQualifiers.add(Default.Literal.INSTANCE);
+            }
```

Comment:
With our recently acquired deep knowledge of the subject, should this be just `if (eventQualifiers.isEmpty())`?

Reply:
Yes, I believe it should.

Reply:
This should now be fixed